### PR TITLE
Feature add docker

### PR DESCRIPTION
**Pull request by montashevych**
# Checklist:

- [ ] Added specs for all new ruby code
- [x] Green rspec run
- [x] Green rubocop
- [x] PR review from teammate

# What?
Added docker instruction for application, deploy app to heroku. 

## Why?
For similar environment for all contributors 

## How?
Created Dockerfile, docker-compose (for fine tuning), heroku file, added postgresql for production environment. 

## Testing?
Did you write tests? nope

## Screenshots (optional)
heroku-container
heroku-postgresql
postgrasql_done

## Anything Else?
yes. For use  you need:
docker-compose build
docker-compose run web rake db:create

for local:
docker-compose up

for deploy on heroku:
heroku container:push web
heroku container:release web
then go to heroku website in your profile and add postgresql in resources.

### CODE REVIEW

**Review by VanyaNovak — APPROVED** [review]
LGTM

File: config/environments/development.rb [hunk 1]
```diff
@@ -11,6 +11,9 @@
   # Do not eager load code on boot.
   config.eager_load = false
 
+  #Blocked Host for dynamic urls. Can change url.
+  config.hosts << "pacific-mountain-49875.herokuapp.com"
```
Comment (andrjuha01) [inline config/environments/development.rb:15]: why it's in development?

File: config/environments/production.rb [hunk 2]
```diff
@@ -52,6 +52,9 @@
   # information to avoid inadvertent exposure of personally identifiable information (PII).
   config.log_level = :info
 
+  #Blocked Host for dynamic urls. Can change url.
+  config.hosts << /[a-z0-9]+\.c9users\.io/
```
Comment (andrjuha01) [inline config/environments/production.rb:56]: should this be set for development?

Comment (montashevych) [reply config/environments/production.rb]: changed. It's really setting only for development environment.

File: config/environments/development.rb (hunk 1 above)
Comment (montashevych) [reply config/environments/development.rb:15]: because this will safe-list the subdomain to access our rails host.

Comment (andrjuha01) [reply config/environments/development.rb:15]: no, I mean why you use heroku domain for development config?
It should be set for production only

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,18 @@
+FROM ruby:3.0.0
+RUN apt-get update -qq && apt-get install -y nodejs postgresql-client
+RUN mkdir /myapp
+ARG RAILS_ENV=production
```
Comment (andrjuha01) [inline Dockerfile:4]: do you need this after this change?
https://github.com/montashevych/Hostmeplease/pull/51/files#diff-b166555f3cff564cc2e83972c0a031baf3cbcd64dd39d6bc509400543ddfae8eR7
looks like duplicate

File: config/environments/production.rb (hunk 2 above)
Comment (andrjuha01) [reply config/environments/production.rb]: should we set production host here?
that one from heroku

File: Dockerfile (hunk 3 above)
Comment (montashevych) [reply Dockerfile]: yep. but i thought if we will push our app to another place (like aws) it's can be helpful. DRY of course)

File: config/environments/development.rb [hunk 4]
```diff
@@ -11,6 +11,7 @@
   # Do not eager load code on boot.
   config.eager_load = false
 
+
```
Comment (andrjuha01) [inline config/environments/development.rb:14]: remove extra new line

Probably need to add heroku host to production.rb?

Comment (montashevych) [reply config/environments/development.rb]: something like this `config.hosts << "pacific-mountain-49875.herokuapp.com" `?
For environments other the development, config.hosts is empty by default. This means no domain checks will be done and the guard will be inactive.

**Review by andrjuha01 — APPROVED** [review]

**Pull request merged**